packages/@aws-cdk/aws-lambda/lib/permission.ts

```diff
@@ -65,3 +65,5 @@ export interface Permission {
    */
   readonly sourceArn?: string;
 }
+
+export const PERMISSION_SUPPORTED_PRINCIPAL_CONDITIONS = [{ operator: 'ArnLike', key: 'aws:SourceArn' }, { operator: 'StringEquals', key: 'aws:SourceAccount' }];
```

Comment:
This is going to be exported publicly. We don't want that.

Just make this a simple const in `function-base.ts`.